scriptworker/test/test_worker.py

```diff
@@ -163,15 +164,22 @@ def test_mocker_run_tasks_noop(context, successful_queue, event_loop, mocker):
     assert status is None
 
 
-@pytest.mark.parametrize("func_to_raise,exc,expected", ((
-    'run_task', ScriptWorkerException, ScriptWorkerException.exit_code
+@pytest.mark.parametrize("func_to_raise,exc,expected,raises", ((
+    'run_task', ScriptWorkerException, ScriptWorkerException.exit_code, False
 ), (
-    'upload_artifacts', ScriptWorkerException, ScriptWorkerException.exit_code
+    'run_task', ValueError, None, True
 ), (
-    'upload_artifacts', aiohttp.ClientError, STATUSES['intermittent-task']
+    'upload_artifacts', ScriptWorkerException, ScriptWorkerException.exit_code,
+    False
+), (
+    'upload_artifacts', aiohttp.ClientError, STATUSES['intermittent-task'],
+    False
+), (
+    'upload_artifacts', OSError, None, True
 )))
-def test_mocker_run_tasks_exception(context, successful_queue, event_loop,
-                                   mocker, func_to_raise, exc, expected):
+@pytest.mark.asyncio
+async def test_mocker_run_tasks_exception(context, successful_queue, mocker,
+                                          func_to_raise, exc, expected, raises):
     """Raise an exception within the run_tasks try/excepts and make sure the
     status is changed
     """
```

Comment:
Looking at this, I'm not sure whether the code or tests is correct. In the unexpected exception case, it appears that we don't change the status.
- If the code is correct, we need to adjust the documentation to reflect the new behavior.
- However, it appears that an unexpected exception will cause the process to exit (good!) but also leave the task claimed (meaning that it will need to timeout the claim). That isn't unreasonable behavior, but we should probably call it out somewhere.
This probably wants to be adjusted to account for unexpected exceptions

Reply:
Yeah, I was thinking about fixing that this PR, and decided to leave it for #131. I do log a special string for the unexpected exception, so we can search for them in papertrail. Does that work?

Reply:
I'm happy to let this be fixed in a separate PR, but then I think the new behavior we are codifying in a test here we think is wrong and are going to correct in a latter PR.

I'm not sure what the solution here is, but I don't think there is value having tests that ossify incorrect behavior.

Reply:
I think the behavior is as designed, with plans to change the behavior later.
I updated the docstring, since I think that's what you're commenting on.